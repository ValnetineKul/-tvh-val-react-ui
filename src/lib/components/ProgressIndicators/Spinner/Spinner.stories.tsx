import { createMeta, createTemplate } from '../../../story-utils';
import Spinner from './Spinner';

const Template = createTemplate(Spinner);

export const Primitive = Template.bind({});
Primitive.storyName = 'Spinner';
Primitive.args = {
  size: 'md',
};

export default createMeta({
  title: 'components/ProgressIndicators/Spinner/Spinner',
  component: Spinner,
  argTypes: {
    spinnerClassName: {
      control: {
        disable: true,
      },
    },
  },
});
