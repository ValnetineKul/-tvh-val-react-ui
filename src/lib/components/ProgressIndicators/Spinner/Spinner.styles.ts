import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles({ name: 'Spinner' })((theme) => ({
  spinnerWrapper: {
    display: 'flex',
  },
  spinner: {
    color: theme.color['accents/onSurfaceCurrent/primary/default'],
  },
}));

export default useStyles;
