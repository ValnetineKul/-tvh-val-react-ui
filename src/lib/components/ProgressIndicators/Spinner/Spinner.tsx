import type { FC } from 'react';
import React from 'react';
import { CircularProgress } from '@mui/material';
import useStyles from './Spinner.styles';

enum SizesEnum {
  'sm' = 12,
  'md' = 19,
  'lg' = 36,
}

type Size = keyof typeof SizesEnum;

export interface SpinnerProps {
  size?: Size;
  className?: string;
  spinnerClassName?: string;
}

const Spinner: FC<SpinnerProps> = ({ size = 'md', className, spinnerClassName }) => {
  const { classes, cx } = useStyles();

  return (
    <div className={cx(classes.spinnerWrapper, className)}>
      <CircularProgress className={cx(classes.spinner, spinnerClassName)} thickness={6.2} size={SizesEnum[size]} />
    </div>
  );
};

export default Spinner;
