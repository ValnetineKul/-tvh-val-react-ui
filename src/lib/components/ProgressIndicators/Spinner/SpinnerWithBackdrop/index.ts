import SpinnerWithBackdrop from './SpinnerWithBackdrop';

export type { SpinnerWithBackdropProps } from './SpinnerWithBackdrop';
export default SpinnerWithBackdrop;
