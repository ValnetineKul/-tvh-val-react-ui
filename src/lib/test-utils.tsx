import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
