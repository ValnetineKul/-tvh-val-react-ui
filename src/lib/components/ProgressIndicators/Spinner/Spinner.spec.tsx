import React from 'react';
import Spinner from './Spinner';
import { render } from '../../../test-utils';

describe('Spinner', () => {
  it('Should render correctly', () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild).toMatchSnapshot('Spinner');
  });
});
