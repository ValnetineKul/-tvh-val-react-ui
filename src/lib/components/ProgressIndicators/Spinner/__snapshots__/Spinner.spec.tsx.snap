// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Should render correctly: Spinner 1`] = `
<div
  class="tss-16l8aab-Spinner-spinnerWrapper"
>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary tss-j8alh2-Spinner-spinner mui-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style="width: 19px; height: 19px;"
  >
    <svg
      class="MuiCircularProgress-svg mui-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate mui-176wh8e-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="18.9"
        stroke-width="6.2"
      />
    </svg>
  </span>
</div>
`;
